public/media_engagement/imaqal_media_engagement.js

```diff
@@ -0,0 +1,455 @@
+const update = (data) => {
+
+    // format the data  
+    data.forEach(function (d) {
+        d.datetime = new Date(d.datetime);
+        d.total_received = +d.total_received
+        d.total_sent = +d.total_sent
+        d.total_pending = +d.total_pending
+        d.total_errored = +d.total_errored
+        d.golis_received= +d.operators["golis"]["received"]
+        d.hormud_received= +d.operators["hormud"]["received"]
+        d.nationlink_received= +d.operators["nationlink"]["received"]
+        d.somnet_received= +d.operators["somnet"]["received"]
+        d.somtel_received= +d.operators["somtel"]["received"]
+        d.telesom_received= +d.operators["telesom"]["received"]
+        d.golis_sent= +d.operators["golis"]["sent"]
+        d.hormud_sent= +d.operators["hormud"]["sent"]
+        d.nationlink_sent= +d.operators["nationlink"]["sent"]
+        d.somnet_sent= +d.operators["somnet"]["sent"]
+        d.somtel_sent= +d.operators["somtel"]["sent"]
+        d.telesom_sent= +d.operators["telesom"]["sent"]
+
+    });
+
+    data.sort((a, b) => new Date(a.datetime) - new Date(b.datetime));
+    
+    // set scale domains
+    x.domain(d3.extent(data, d => new Date(d.datetime)));
+    y_total_received_sms.domain([0, d3.max(data, function (d) { return Math.max(d.total_received, d.golis_received, d.hormud_received, d.nationlink_received, d.somnet_received, d.somtel_received, d.telesom_received); })]);
+    y_total_sent_sms.domain([0, d3.max(data, function (d) { return Math.max(d.total_sent, d.golis_sent, d.hormud_sent, d.nationlink_sent, d.somnet_sent, d.somtel_sent, d.telesom_sent); })]);
+    y_total_failed_sms.domain([0, d3.max(data, function (d) { return Math.max(d.total_errored); })]);
+
+    // update path data for total incoming sms(s)
+    total_received_path.data([data])
+        .attr("class", "line")
+        .style("stroke", "green")
+        .attr("d", total_received_line);
+    total_golis_received_path.data([data])
+        .attr("class", "line")
+        .style("stroke", "#FF6F61")
+        .attr("d", total_golis_received_line);
+    total_hormud_received_path.data([data])
+        .attr("class", "line")
+        .style("stroke", "#6B5B95")
+        .attr("d", total_hormud_received_line);
+    total_nationlink_received_path.data([data])
+        .attr("class", "line")
+        .style("stroke", "#790303")
+        .attr("d", total_nationlink_received_line);
+    total_somnet_received_path.data([data])
+        .attr("class", "line")
+        .style("stroke", "#5A3E36")
+        .attr("d", total_somnet_received_line);
+    total_somtel_received_path.data([data])
+        .attr("class", "line")
+        .style("stroke", "#935529")
+        .attr("d", total_somtel_received_line);
+    total_telesom_received_path.data([data])
+        .attr("class", "line")
+        .style("stroke", "#2A4B7C")
+        .attr("d", total_telesom_received_line);
+    
+    // update path data for total outgoing sms(s)
+    total_sent_path.data([data])
```

Comment:
As discussed lets factor into a dictionary

Reply:
Thanks will refactor in a future pr.